### PR TITLE
Padding mask for varying length trajectories

### PR DESCRIPTION
**Pull request by han-ol**
Addresses #713.

### CODE REVIEW

File: bayesflow/networks/summary/transformers/time_series_transformer.py [hunk 1]
```diff
@@ -139,9 +146,16 @@ def call(self, x: Tensor, training: bool = False, attention_mask: Tensor = None)
         if self.time_embedding:
             inp = self.time_embedding(inp, t=t)
 
+        if attention_mask is None and mask is not None:
+            attention_mask = keras.ops.expand_dims(keras.ops.cast(mask, "bool"), axis=1)
```
Comment (stefanradev93) [inline bayesflow/networks/summary/transformers/time_series_transformer.py:150]: As per keras doc, the attention mask should have a shape of `(B, T, S=T)`, no valid broadcasting is indicated for shapes `(B, 1, T)`.

Comment (stefanradev93) [reply bayesflow/networks/summary/transformers/time_series_transformer.py:150]: Bump.

Comment (stefanradev93) [reply bayesflow/networks/summary/transformers/time_series_transformer.py:150]: I confirm that the attention mask properly broadcasts with `(B, 1, T)`.

Comment (han-ol) [reply bayesflow/networks/summary/transformers/time_series_transformer.py:150]: Great, thanks
